py/desispec/workflow/procfuncs.py

```diff
@@ -304,7 +304,8 @@ def create_batch_script(prow, queue='realtime', dry_run=0, joint=False, system_n
     if prow['JOBDESC'] in ['perexp','pernight','pernight-v0','cumulative']:
         if dry_run > 1:
             scriptpathname = get_tile_redshift_script_pathname(tileid=prow['TILEID'],group=prow['JOBDESC'],
-                                                               night=prow['NIGHT'], expid=prow['EXPID'])
+                                                               night=prow['NIGHT'], expid=prow['EXPID'][0])
```

Comment:
@akremin previously this step was crashing with dry_run=2 since prow['EXPID'] is an array of EXPIDs not just a single EXPID (get_tile_redshift_script_pathname wants EXPID=int, not an array).
This hack isn't quite right either since JOBDESC='perexp' would submit several jobs instead of just a job for the first EXPID in the array.

I patched what I needed to move on with testing the actual PR, but I'm highlighting this for you in case there is a more-correct simple workaround?

Reply:
Great catch. I'm not sure how my initial vetting missed this, I must not have tested all 3 dry_runs * 4 redshifts = 12 combinations. Once you're happy with the PR, I'll piggy back on it and add a complete fix for this so that it works properly.